core/src/main/java/com/linecorp/armeria/internal/server/servlet/ServletTlsAttributes.java

```diff
@@ -108,7 +111,8 @@ private static ServletTlsAttributes getOrCreateAttrs(SSLSession session) {
             return attrs;
         }
 
-        synchronized (session) {
+        lock.lock();
```

Comment:
I would actually argue that it would be better revert changes in this class and use `synchronized` since:

1) The lock is held statically across connections instead of within a single connection (since a `SSLSession` is created per connection), leading to more unnecessary contention.
2) There aren't any io/native calls in this block subject to [pinning](https://cr.openjdk.java.net/~rpressler/loom/loom/sol1_part1.html#pinning)